src/connection.rs

```diff
@@ -420,45 +429,47 @@ where
 {
     let mut state = State::ServerVerify;
     decrypt_record(key_schedule.read_state(), record, |key_schedule, record| {
-        debug!("record = {:?}", record.content_type());
+        trace!("record = {:?}", record.content_type());
         match record {
-            ServerRecord::Handshake(server_handshake) => match server_handshake {
-                ServerHandshake::EncryptedExtensions(_) => {}
-                ServerHandshake::Certificate(certificate) => {
-                    trace!("Verifying certificate!");
-                    let transcript = key_schedule.transcript_hash();
-                    handshake
-                        .verifier
-                        .verify_certificate(transcript, &config.ca, certificate)?;
-                    trace!("Certificate verified!");
-                }
-                ServerHandshake::CertificateVerify(verify) => {
-                    trace!("Verifying signature!");
-                    handshake.verifier.verify_signature(verify)?;
-                    trace!("Signature verified!");
-                }
-                ServerHandshake::CertificateRequest(request) => {
-                    trace!("Certificate requested");
-                    handshake.certificate_request.replace(request.try_into()?);
-                }
-                ServerHandshake::Finished(finished) => {
-                    trace!("************* Finished");
-                    if !key_schedule.verify_server_finished(&finished)? {
-                        return Err(TlsError::InvalidSignature);
+            ServerRecord::Handshake(server_handshake) => {
+                trace!("handshake = {:?}", server_handshake.handshake_type());
```

Comment:
This whole diff is just formatting change because the trace! requires a set of braces around the match.